File: xskeleton-lib/src/main/java/com/mcgrady/xskeleton/integration/IRepositoryManager.java
package com.mcgrady.xskeleton.integration;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import retrofit2.Retrofit;

/**
 * Created by mcgrady on 2019-08-10.
 */
public interface IRepositoryManager {

    /**
     * 根据传入的 Class 获取对应的 Retrofit service
     *
     * @param service Retrofit service class
     * @param <T>     Retrofit service 类型
     * @return Retrofit service
     */
    @NonNull
    <T> T obtainRetrofitService(@NonNull Class<T> service);

    /**
     * 根据传入的 Class 获取对应的 RxCache service
     *
     * @param cache RxCache service class
     * @param <T>   RxCache service 类型
     * @return RxCache service
     */
    @NonNull
    <T> T obtainCacheService(@NonNull Class<T> cache);

    /**
     * 清理所有缓存
     */
    void clearAllCache();

    /**
     * 获取 {@link Context}
     *
     * @return {@link Context}
     */
    @NonNull
    Context getContext();

    interface ObtainServiceDelegate {
        @Nullable
        <T> T createRetrofitService(Retrofit retrofit, Class<T> serviceClass);
    }
}

File: xskeleton-lib/src/main/java/com/mcgrady/xskeleton/integration/IRepositoryManager.kt
package com.mcgrady.xskeleton.integration

import com.mcgrady.xskeleton.http.handler.RxErrorHandler

/**
 * Created by mcgrady on 2019-08-10.
 */
interface IRepositoryManager {
    /**
     * 根据传入的 Class 获取对应的 Retrofit service
     *
     * @param service Retrofit service class
     * @param <T>     Retrofit service 类型
     * @return Retrofit service
    </T> */
    fun <T> obtainRetrofitService(service: Class<T>): T

    val rxErrorHandler: RxErrorHandler?
}